rabbitmq/publisher.go

```diff
@@ -54,29 +53,65 @@ func NewRabbitMqPublisher(args ArgsRabbitMqPublisher) (*rabbitMqPublisher, error
 		cfg:                args.Config,
 		client:             args.Client,
 		closeChan:          make(chan struct{}),
-	}, nil
+	}
+
+	err = rp.checkAndCreateExchanges()
+	if err != nil {
+		return nil, err
+	}
+
+	return rp, nil
 }
 
 func checkArgs(args ArgsRabbitMqPublisher) error {
 	if check.IfNil(args.Client) {
 		return ErrNilRabbitMqClient
 	}
-	if args.Config.EventsExchange == "" {
-		return fmt.Errorf("empty events exchange name: %w", ErrInvalidRabbitMqExchangeName)
+
+	return nil
+}
+
+// checkAndCreateExchanges creates exchanges if they are not existing already
+func (rp *rabbitMqPublisher) checkAndCreateExchanges() error {
+	err := rp.checkAndCreateExchange(rp.cfg.EventsExchange)
+	if err != nil {
+		return err
 	}
-	if args.Config.RevertEventsExchange == "" {
-		return fmt.Errorf("empty revert exchange name: %w", ErrInvalidRabbitMqExchangeName)
+	err = rp.checkAndCreateExchange(rp.cfg.RevertEventsExchange)
+	if err != nil {
+		return err
 	}
-	if args.Config.FinalizedEventsExchange == "" {
-		return fmt.Errorf("empty finalized exchange name: %w", ErrInvalidRabbitMqExchangeName)
+	err = rp.checkAndCreateExchange(rp.cfg.FinalizedEventsExchange)
+	if err != nil {
+		return err
 	}
-	if args.Config.BlockTxsExchange == "" {
-		return fmt.Errorf("empty block txs exchange name: %w", ErrInvalidRabbitMqExchangeName)
+	err = rp.checkAndCreateExchange(rp.cfg.BlockTxsExchange)
+	if err != nil {
+		return err
+	}
+	err = rp.checkAndCreateExchange(rp.cfg.BlockScrsExchange)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (rp *rabbitMqPublisher) checkAndCreateExchange(conf config.RabbitMQExchangeConfig) error {
```

Comment:
this function does 2 things: split maybe?

Reply:
splitted check and create